libraries/AP_Compass/CompassCalibrator.cpp

```diff
@@ -393,6 +394,29 @@ void CompassCalibrator::thin_samples() {
     update_completion_mask();
 }
 
+bool CompassCalibrator::accept_sample(const Vector3f& sample, const uint16_t idx)
```

Comment:
could you add a comment to this function explaining what it is doing?

Reply:
This is just a copy of the existing accept_sample which exempts an item from the distance check.

It would better be done with an optional `int32_t idx` with a magic value of `-1` meaning don't exempt any element from the check.

Reply:
I just made a simple commit which I would thought it could be somewhat a litter clear for you guys to understand what I mean to mention in my PR. I then would push a better commit to this PR based in @peterbarker mentioned above.